src/aind_dynamic_foraging_basic_analysis/metrics/session_metrics.py

```diff
@@ -0,0 +1,217 @@
+
+    p_larger_global = max(
+        df["probability_chosen"].max(), df["probability_not_chosen"].max()
+    )
+
+    p_smaller_global = min(
+        df["probability_chosen"].min(), df["probability_not_chosen"].min()
+    )
+
+    mean_difference = average.mean()
+    chosen_probability = mean_difference / (p_larger_global - p_smaller_global)
+
+    # Pack all data
+    dict_meta = {
+        # Basic metadata
+        "rig": meta_dict["box"],
+        "user_name": nwb.experimenter[0],
+        "task": nwb.protocol,
+        # New metric
+        "chosen_probability": chosen_probability,
+        # Trial counts and rates
+        "total_trials": n_total_trials_non_autowater,
+        "finished_trials": n_finished_trials_non_autowater,
+        "finished_rate": finished_rate,
+        "total_trials_with_autowater": n_total_trials,
+        "finished_trials_with_autowater": n_finished_trials,
+        "finished_rate_with_autowater": n_finished_trials / n_total_trials,
+        # Reward and foraging metrics
+        "reward_trials": n_reward_trials_non_autowater,
+        "reward_rate": reward_rate_non_autowater_finished,
+        "foraging_eff": foraging_eff,
+        "foraging_eff_random_seed": foraging_eff_random_seed,
+        "foraging_performance": foraging_eff * finished_rate,
+        "foraging_performance_random_seed": foraging_eff_random_seed * finished_rate,
+        # Timing metrics
+        "reaction_time_median": df.loc[:, "reaction_time"].median(),
+        "reaction_time_mean": df.loc[:, "reaction_time"].mean(),
+        "early_lick_rate": (df.loc[:, "n_lick_all_delay_period"] > 0).sum() / n_total_trials,
+        # Double dipping metrics
+        "double_dipping_rate_finished_trials": (
+            df.loc[(df.animal_response != IGNORE), "n_lick_switches_gocue_stop"] > 0
+        ).sum()
+        / (df.animal_response != IGNORE).sum(),
+        "double_dipping_rate_finished_reward_trials": (
+            df.loc[df.reward, "n_lick_switches_gocue_stop"] > 0
+        ).sum()
+        / df.reward.sum(),
+        "double_dipping_rate_finished_noreward_trials": (
+            df.loc[
+                (df.animal_response != IGNORE) & (~df.reward),
+                "n_lick_switches_gocue_stop",
+            ]
+            > 0
+        ).sum()
+        / ((df.animal_response != IGNORE) & (~df.reward)).sum(),
+        # Lick consistency metrics
+        "lick_consistency_mean_finished_trials": df.loc[
+            (df.animal_response != IGNORE), "n_lick_consistency_gocue_stop"
+        ].mean(),
+        "lick_consistency_mean_finished_reward_trials": df.loc[
+            df.reward, "n_lick_consistency_gocue_stop"
+        ].mean(),
+        "lick_consistency_mean_finished_noreward_trials": df.loc[
+            (df.animal_response != IGNORE) & (~df.reward),
+            "n_lick_consistency_gocue_stop",
+        ].mean(),
+    }
+
+    # Create DataFrame with hierarchical columns
```

Comment:
Is this something you inherited from `process_nwbs.py`? I'm not sure we need it

Reply:
Yes, but I guess it's not necessary if we're not using session indices?